### DISCUSSION TITLE
Library GUIslice can't compile Mega2560 and TFT 3.5" ILI9486

### DISCUSSION THREAD

**Discussion by ChatIbex** [ImpulseAdventure/GUIslice#399] [Q&A]
Dear Sir,

We have load library GUIslice. It can not compile board Mega2560 and TFT 3.5" ILI9486. Please advise to me.

BestRegards,
Pariphat

**Comment by Pconti31** [comment]
@ChatIbex 
1. For using any display always start with reading the GUIslice wiki on configuring.
[GUIslice Wiki](https://github.com/ImpulseAdventure/GUIslice/wiki)
2. Make your Arduino and Display work with the display driver without GUIslice.  ILI9486 isn't supported by Adafruit's GFX so in your case you might choose to use the [MCUFRIEND_kbv library. ](https://github.com/prenticedavid/MCUFRIEND_kbv)
3.  Once your display and mega works with this library you can start to work with GUIslice.  Again refer to the wiki. If you get errors post in the issue section of GUIslice not discussions so you can post the compiler errors and a zip file of your sample project, with your GUIslice/config/ard-shld-mcufriend.h, or ard-shld-mcufriend_4wire.h or ard-shld-mcufriend_xpt2046.h or whatever config file you used and your GUIslice/src/GUIslice_config.h file
4. DO NOT forget to also post what version number of GUIslice you are using and maybe a link to your display.

Paul--

**Reply by ImpulseAdventure** [reply]
To add to Paul’s recommendation above, it is really important for us to see the exact error message that you are getting during compile, in addition to the version (you will see this in /src/GUIslice_version.h).

**Comment by ChatIbex** [comment]
Dear Sir,

We are new to arduino we have display tft 3.5” driver :ILI9486 Bus :8 bit dot: 320x480 SKU: MAR3502 with arduino mega2560 add library you GUIslice-master  Preview and verify failed.  Please guide us just getting started with arduino.

Mr.Pariphat Amornsupparat

**Reply by Pconti31** [reply]
@ChatIbex Sorry, but my guidance above still holds.  You need to follow it step by step 1 through 4. Your Display is 8-bit parallel interface which is only supported by
[MCUFRIEND_kbv Library](https://github.com/prenticedavid/MCUFRIEND_kbv). 

Either load this library from inside Arduino IDE or download by clicking link above (the blue letters MCUFRIEND_kbv).  Get the MCUFREIND_kbv samples to work and only then worry about GUIslice.   Trouble? Ask for help on the MCUFRIEND_kbv issues part of David Prentice's GITHUB repository.

NOTE: The MAR3502 does not support touch input!  So you will need to wire your own buttons or buy the MAR3501 version of this display.  

If none of this makes sense to you begin with asking questions on the [Arduino Forum](https://forum.arduino.cc/) (again click on the blue letters) which is set up for beginners.
Paul--
